### ISSUE TITLE
Floccus 3.2.0: Error when deleting bookmarks

### ISSUE THREAD

**Issue by gohrner** [marcelklehr/floccus#216]
![image](https://user-images.githubusercontent.com/26199042/50358705-8ea6c380-055a-11e9-9ec3-ee123bde7576.png)

**Comment by gohrner** [comment]
Syncing does not work any more at all afterwards, I keep getting this error message even after a browser restart.

Is there any way to get Floccus working again besides removing the account and sync from scratch?

**Comment by marcelklehr** [comment]
Mmh. Ticking the 'Sync from scratch' option to reset the cache might work. Can you give it a try?

**Comment by marcelklehr** [comment]
@gohrner This is most probably due to the fact that you still have the old floccus version on computer B. Mozilla's addon review team seem to be on holiday at the moment :/

**Comment by gohrner** [comment]
Ok, then I'll just wait for the update to show up and test again. Then I'll just close this bug as a "hickup"?

**Comment by gohrner** [comment]
Meanwhile the Firefox addon also got its update to 3.2.1. A first try to synchronize lead to the error message:

Folder ordering is missing some of the folder's children

Afterwards the "Trashbin" folder, which had been deleted in Chrome an in the XBEL file on the server, reappeared. (But the error message vanished.)

I deleted the "Trasbin" folder again and we'll see what happens...

**Comment by marcelklehr** [comment]
> I deleted the "Trasbin" folder again and we'll see what happens...

Did it come back?

**Comment by gohrner** [comment]
Unfortunately yes - it reappeared in Firefox. We then pressed "sync" there to see what happens, and now Chrome throws an Error on sync again: "Folder ordering is missing some of the folder's children"... The Chrome Add-On currently is on 3.2.2, I cannot check the FF addon version at the moment, but it's at least 3.2.1 or also 3.2.2.

**Comment by marcelklehr** [comment]
Happy new year! :tada:

> Unfortunately yes - it reappeared in Firefox. 
Debug logs please :)

**Comment by gohrner** [comment]
Sent logs by mail.

**Comment by gohrner** [comment]
Must be a demonstration effect, but finally the folder also got deleted on the Firefox side and is still gone in Chrome... Must be the demonstration effect. I didn't do anything further besides waiting.

**Comment by marcelklehr** [comment]
hah, I think I was able to reproduce this, though. Let's see if the next version fixes things for you as well.

**Comment by gohrner** [comment]
Great, I'm curious. :)

**Comment by erdnussflips** [comment]
I'm getting also "Folder ordering is missing some of the folder's children".
Waiting for the next version. :)

**Comment by marcelklehr** [comment]
The new version is released.

**Comment by erdnussflips** [comment]
For me, the new version doesn't fix the problem. Still getting "Folder ordering is missing some of the folder's children". :(

**Comment by marcelklehr** [comment]
Ooph. This bug is tough.

So, according to @gohrner the general pattern is the following:

* Client A: delete folder and trigger sync
* => folder is deleted in client A and the XBEL file
* Client B: syncing outcome: folder is not deleted and the error "Cannot find parent bookmark for id" is shown
* Client B: Syncing again causes folder to be added to the XBEL file again
* Client A: Sync outcome: "Folder ordering is missing some of the folder's children"

At the moment @erdnussflips seems to be at the last step. I think, it is possible that the last step persists even though the original bug is fixed. Could you reset the cache on both clients and thus start from scratch? The folder in question will likely be created on all clients again, but then it'd be interesting to see if the first error reappears and if any other errors occur at all.

**Comment by marcelklehr** [comment]
Any news on this?

**Comment by gohrner** [comment]
Just noticed that Floccus apparently lost my account completely sometime recently - so no wonder I did not see any sync failures. ;) No account is configured at all at the moment... :-/

(The sync failures also seem to have been more or less resolved by your previous bugfix releases.)

**Comment by marcelklehr** [comment]
This is all very strange. :thinking: 

How did it go on your end @erdnussflips?

**Comment by erdnussflips** [comment]
Hi @marcelklehr, i reseted both clients but, the error still appeared. 

At the end I exported my bookmarks through the browser. Then I removed all the bookmarks from all synced browsers, reseted the cache and let floccus sync the empty bookmark list.

After that I reimported all my bookmarks in browser 1 and started floccus sync. Then I started floccus sync in Browser 2. Now the problem is gone.
